fragment.go

```diff
@@ -1668,11 +1671,15 @@ func (f *fragment) bulkImportMutex(rowIDs, columnIDs []uint64) error {
 		if err != nil {
 			return err
 		}
-		rowIDs[setIdx] = pos
-		setIdx++
+		colSet[columnID] = pos
 		rowSet[rowID] = struct{}{}
 	}
-	toSet := rowIDs[:setIdx]
+	i := 0
+	for _, pos := range colSet {
+		rowIDs[i] = pos
```

Comment:
this confused the heck out of me for a while but i think i follow it. i have a vague feeling of impending doom but i think it's actually correct. only thing i notice is that if you had multiple writes to a column, you'd end up generating multiple clearbits of the same existing bit for it, but that doesn't break anything, and it's probably cheaper than setting up a thing parallel to the colSet tracking.